### PR TITLE
fix: handle content from loaders

### PR DESCRIPTION
**Pull request by alexander-akait**
**What kind of change does this PR introduce?**

bugfix

Unfortunately, we cannot avoid encode data URI after loaders, loader can change content and encoding will be same, so we need to generate data URI from source again and we can't reuse `encodedContent`, bad - it generates more longer data URIs

**Did you add tests for your changes?**

Yes

**Does this PR introduce a breaking change?**

No

**What needs to be documented once your changes are merged?**

No

### CODE REVIEW

File: lib/asset/AssetGenerator.js [hunk 1]
```diff
@@ -157,35 +185,7 @@ class AssetGenerator extends Generator {
 							);
 						}
 
-						let encodedContent;
-						if (
-							module.resourceResolveData &&
-							module.resourceResolveData.encoding === encoding
```
Comment (sokra) [inline lib/asset/AssetGenerator.js:163]: I think I would also work to add a check here to check if the content is still equal: `decodeDataUriContent(module.resourceResolveData.encodedContent, module.resourceResolveData.encoding).equals(originalSource.buffer())`

So we keep encoding intact if possible.

**Review by sokra — CHANGES_REQUESTED** [review]

File: lib/asset/AssetGenerator.js (hunk 1 above)
Comment (sokra) [reply lib/asset/AssetGenerator.js:163]: They must be equal when no loader transformation has happened

Comment (alexander-akait) [reply lib/asset/AssetGenerator.js:163]: @sokra Fixed

**Review by sokra — APPROVED** [review]

**Comment by sokra** [comment]
Thanks

**Pull request merged**